src/facade/cmd_arg_parser_test.cc

```diff
@@ -95,31 +96,42 @@ TEST_F(CmdArgParserTest, NextStatement) {
 }
 
 TEST_F(CmdArgParserTest, CheckTailFail) {
-  auto parser = Make({"TAG", "11", "22", "TAG", "33"});
+  auto parser = Make({"TAG", "11", "22", "TAG", "text"});
 
-  EXPECT_TRUE(parser.Check("TAG"));
-  parser.Skip(2);
-
-  EXPECT_TRUE(parser.Check("TAG"));
-  parser.Next<int, int>();
+  int first;
+  string_view second;
+  EXPECT_TRUE(parser.Check("TAG", &first, &second));
+  EXPECT_EQ(first, 11);
+  EXPECT_EQ(second, "22");
 
-  auto err = parser.Error();
-  EXPECT_TRUE(err);
-  EXPECT_EQ(err->index, 4);
+  EXPECT_FALSE(parser.Check("TAG", &first, &second));
+  EXPECT_TRUE(parser.Check("TAG", &first));
```

Comment:
why do we return true here?

Reply:
because after TAG we have one string field so we can "TAG" and "text"
in the previous line, we try to consume more elements than exist so we return false without consuming